metric_map.go

```diff
@@ -367,6 +373,7 @@ func (mm *MetricMap) String() string {
 }
 
 // AsMetrics will synthesize Metrics from the MetricMap and return them as a slice
+// TODO: Remove this function, it only supports tests now
```

Comment:
I'll raise an issue after this is merged.

Reply:
Sure.  Does your IDE use `deprecated` or `Deprecated`? Goland seems to want the D or it won't highlight it.

Reply:
Goland has a "deprecate harder", for when the symbol is actually intended to be removed, but damned if I can find the comment syntax for that.